File: Semester_3/Operating_System/Examples/hello/hello.go
/*
 * hello.go --
 *
 *	A go hello world program, which uses a goroutine and a
 *	channel to block the main function. Build an executable
 *	using 'go build hello.go'.
 */
 
package main

import "fmt"

func hello(d chan<- bool) {
	fmt.Println("hello world")
 	d <- true
}

func main() {
        d := make(chan bool)
	go hello(d)
        <- d // block on a channel that indicates that the hello() is done
}

File: semester_3/operating_system/src/c/hello/hello.rs
/*
 * hello.rs --
 *
 *	A rust hello world program, which uses a thread to print
 *	hello world. Build an executable using:
 *
 *	rustc -o hello -O hello.rs
 */
 
use std::thread;

fn main() {
    let w = "world";

    let handle = thread::spawn(move || {
        println!("Hello {}!", w);
    });

    handle.join().unwrap();
}
